### PR TITLE
Add open tracing to pulsar go clinet

### PR DESCRIPTION
**Pull request by maraiskruger1980**
As suggested by @sijie
image

Here is the PR to add open tracing to the pulsar go client directly.
This is replicating what the java tracer does: 
https://github.com/streamnative/pulsar-tracing
The usage is stipulated in the readme.md file

### CODE REVIEW

**Comment by wolfstudy** [comment]
@maraiskruger1980 Great Job., thanks for your contribution. Can you add some test cases for this?

It seems that we can also add similar functions to the reader. Are you also interested in this?

**Review by wolfstudy — COMMENTED** [review]
@maraiskruger1980 Great Job., thanks for your contribution. Can you add some test cases for this?

It seems that we can also add similar functions to the reader. Are you also interested in this?

**Comment by wolfstudy** [comment]
> It seems that we can also add similar functions to the reader. Are you also interested in this?

For this, you can also send a new pull request to impl it.

**Comment by maraiskruger1980** [comment]
> @maraiskruger1980 Great Job., thanks for your contribution. Can you add some test cases for this?
> 
> It seems that we can also add similar functions to the reader. Are you also interested in this?

@wolfstudy I added tests and made some improvements

**Review by wolfstudy — APPROVED** [review]
LGTM +1

**Review by wolfstudy — APPROVED** [review]

**Comment by wolfstudy** [comment]
Here is the PR to add open tracing to the pulsar go client directly.
This is replicating what the java tracer does: 
https://github.com/streamnative/pulsar-tracing
The usage is stipulated in the readme.md file

File: pulsar/internal/pulsartracing/readme.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+### Usage
```
Comment (Anonymitaet) [inline pulsar/internal/pulsartracing/readme.md:1]: @maraiskruger1980 thanks for your contribution! Want users to know this great feature and know how to use it? 

If so, could you please add docs to [Pulsar - Go client](https://pulsar.apache.org/docs/en/next/client-libraries-go/)?
(How about adding a section "Monitoring" and add the docs under that section? Explanations of "What is the feature" & "How to use the feature" are needed. You can take [How to trace Pulsar messages with OpenTracing and Jaeger](https://streamnative.io/en/blog/tech/2020-06-11-opentracing-instrumentation-for-pulsar/) and [OpenTracing Pulsar Client](https://hub.streamnative.io/monitoring/opentracing-pulsar-client/0.1.0) as references. 

Then I would like to help review. Feel free to ping me if needed, thanks.

**Pull request merged**